### ISSUE TITLE
Ctrl + arrow keys insert garbage

### ISSUE THREAD

**Issue by KevinSjoberg** [adsr/mle#34]
At times, I type commands really fast and forget to release `ctrl` before pressing an arrow key. Whenever this happens, "garbage" is being inserted in the buffer. Perhaps the respective key code?

- `C-Up` inserts `1;5A`
- `C-Down` inserts `1;5B`
- `C-Left` inserts `1;5D`
- `C-right` inserts `1;5C`

Similar to `Shift-Tab`, the "garbage" is being inserted at nearest `{`. Continuous presses will walk up the tree, inserting the "garbage" at every `{` it can find.

**Comment by adsr** [comment]
This and #33 used to bug me as well. Termbox does not support these key combinations unfortunately. It is one of the reasons (alongside 256-bit color and mouse support) why there are a lot of termbox forks out there[0][1][2]. Although I have no personal desire for high color or mouse support, many people do, and handling extra key combinations like this would be nice.

The termbox maintainer has had the project in maintenance mode for a long while, so we may not get new features any time soon. It is also worth mentioning that I have packaged termbox as a dep of mle on several distros already so it would be painful to switch libraries.

That said I have been thinking about the fragmented state of termbox, and I have a rough idea for a middle ground that might be acceptable to all parties: I propose adding one or two functions to the termbox API which would accept function pointers for extending event parsing and escape codes. I'd like to do a survey of the forks to understand what features have been added before proposing it to a wider audience.

[0] https://github.com/tomas/termbox
[1] https://github.com/cylgom/termbox_next
[2] https://github.com/nsf/termbox/network/members
